app/src/main/java/ca/unb/mobiledev/reflexrevolution/instructions/TypeInstruction.java

```diff
@@ -0,0 +1,128 @@
+    private long fileLength;
+    private final int SIZE_OF_FILE_LINE = 7;
+    private InputMethodManager keyboardDisplayManager;
+
+    public TypeInstruction(ViewGroup layout, Callback callback) {
+        super(layout, callback);
+        rand = new Random();
+        keyboardDisplayManager = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
+        fileLength = 0;
+        setUpRandomWordList();
+    }
+
+    //Adapted from https://stackoverflow.com/a/48865091
+    private void setUpRandomWordList(){
+        try {
+            File outputDir = context.getCacheDir();
+            File destFile = new File(outputDir, "TypeWords.txt");
+            destFile.delete();
+            OutputStream dest = new FileOutputStream(destFile, true);
+            InputStream src = context.getAssets().open("TypeWords.txt", AssetManager.ACCESS_STREAMING);
+            byte[] buff = new byte[100 * 1024];
+            for (; ; ) {
+                int cnt = src.read(buff);
+                if (cnt <= 0)
+                    break;
+                dest.write(buff, 0, cnt);
+            }
+            dest.flush();
+            dest.close();
+            wordListFile = new RandomAccessFile(destFile, "r");
+
+            //Get number of lines in file
+            fileLength = wordListFile.length();
+        } catch(IOException e){ e.printStackTrace(); }
+    }
+
+    @Override
+    public void init() {
+        super.init();
+        //Get random word to type
+        try {
+            //Subtracting file line size outside random so there's an equal chance of getting the first string in the list as the others
+            int pos = (rand.nextInt((int)fileLength) - SIZE_OF_FILE_LINE);
+            if(pos < 0) pos = 0;
+            wordListFile.seek(pos);
+            if(pos != 0) wordListFile.readLine();
+            word = wordListFile.readLine();
+        } catch (IOException e) { e.printStackTrace(); }
+    }
+
+    @Override
+    public void display() {
+        addTextView(word, SMALL_TEXT_SIZE);
+        addTextView("TYPE", DEFAULT_TEXT_SIZE);
+        field = new EditText(context);
+
+        //Listener that checks for text being updated
+        field.addTextChangedListener(new TextWatcher() {
+            //If text equals current word, trigger success
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                if(s.toString().toUpperCase().equals(word)){
+                    //Hide keyboard
+                    disable();
+                    success();
+                }
+            }
+
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+                //Auto-generated
+            }
+            @Override
+            public void afterTextChanged(Editable s) {
+                //Auto-generated
+            }
+        });
+        addView(field);
+        enable();
+    }
```

Comment:
We shouldn't call `enable()` and `disable()` since the GameActivity is already doing it as needed. You should remove all the enable and disable calls from the Type and Dial instructions.

Reply:
Oh yeah... I forgot those would get called in the places I needed them anyway